aries/aries_vcx_core/src/wallet/utils.rs

```diff
@@ -0,0 +1,52 @@
+use public_key::Key;
+use rand::{distributions::Alphanumeric, Rng};
+use serde::Deserialize;
+
+use crate::errors::error::{AriesVcxCoreError, AriesVcxCoreErrorKind, VcxCoreResult};
+
+pub fn bytes_to_string(vec: Vec<u8>) -> VcxCoreResult<String> {
+    String::from_utf8(vec)
+        .map_err(|err| AriesVcxCoreError::from_msg(AriesVcxCoreErrorKind::InvalidInput, err))
+}
+
+pub fn bs58_to_bytes(key: &str) -> VcxCoreResult<Vec<u8>> {
+    bs58::decode(key)
+        .into_vec()
+        .map_err(|err| AriesVcxCoreError::from_msg(AriesVcxCoreErrorKind::WalletError, err))
+}
+
+pub fn bytes_to_bs58(bytes: &[u8]) -> String {
+    bs58::encode(bytes).into_string()
+}
+
+pub fn from_json_str<T: for<'a> Deserialize<'a>>(json: &str) -> VcxCoreResult<T> {
+    serde_json::from_str::<T>(json)
+        .map_err(|err| AriesVcxCoreError::from_msg(AriesVcxCoreErrorKind::InvalidJson, err))
+}
+
+pub fn random_seed() -> String {
```

Comment:
Does not need to be public.

Reply:
You will not get warnings because a function is not used in the file it is defined in (which doesn't make sense). You get warnings for some feature flag combinations when not compiling tests because for those features this function is used only in test code, but it is not hidden under `#[cfg(test)]`. So allowing `dead_code` is not a solution because this warning is warranted.

Reply:
You are right, this is not caused because the function is not called in the same file.

The function is called in code for askar and in tests, so for `cargo clippy -p aries_vcx --features legacy_proof --features credx,vdrtools_wallet` it is never used.